doc/getting_started.rst

```diff
@@ -235,7 +235,45 @@ Now you can use **Lua Fun**:
     2 c
     one
 
-Please note that functions support multireturn.
+.. _multi_return:
+
+Please note that functions support multireturn - many functions return a
+triplet ``gen``, ``param`` and ``state``, so called :ref:`iterator triplet
+<iterator_triplet>`.
+
+Please note that functions support returns of two iterators. Example of these
+functions are :func:`partition` and :func:`span`. One can call ``:unwrap()``
+on iterators to pass them into the ``for <..> in <...>`` loop. Example:
```

Comment:
Will not 'one can call `:unwrap()` on iterators to pass then into `for <..> in <..>` loop' look as a general note instead of be a recipe specifically for functions that return two iterators?

To be honest, I would move accents here and would write something like the following.

<details>
<summary>Example of wording</summary>

> Please note that luafun functions employ multireturn feature of Lua language and return `gen`, `param`, `state` :ref:\`triplet <iterator_triplet>\`. This is exactly what a \`\`for\`\` loop expects, so luafun functions can be used in the place where \`\`pairs\`\` or \`\`ipairs\`\` is usually used:
>
> ```lua
> -- Usual usage.
> > for i in range(3) do print(i) end
> 1
> 2
> 3
> -- The same, but in the more explicit form.
> > gen, param, state = range(3)
> > for i in gen, param, state do print(i) end
> 1
> 2
> 3
> ```
>
> The main practical idea here is that the following will be incorrect:
>
> ```lua
> > it = range(3)
> > for i in it do print(i) end -- error
> ```
>
> However \`\`param\`\` and \`\`state\`\` are actually wrapped inside the \`\`it\`\` iterator and it is possible to unwrap them back to \`\`gen\`\`, \`\`param\`\`, \`\`state\`\`:
>
> ```lua
> -- Explicit form.
> > it = range(3)
> > gen, param, state = it:unwrap()
> > for i in gen, param, state do print(i) end
> 1
> 2
> 3
> -- Usual simple form.
> > it = range(3)
> > for i in it:unwrap() do print(i) end
> 1
> 2
> 3
> ```
>
> Usually it is not important: just call `:totable()` or pass the returned triplet into a \`\`for\`\` loop. But the \`\`unwrap()\`\` trick is very useful for luafun functions, which return two iterators (such as :func:\`partition\` and :func:\`span\`). Consider the example:
>
> \<...\>
</details>

BTW, just observed that the `unwrap()` function/method is not documented.

Do you see my idea? 'luafun supports multireturn' means nothing, the whole Lua language supports it. The practical sense is that we can use the return values in a `for` loop directly. And that we should save them into three variables to pass somewhere or should call `unwrap()`. Next, 'one can call `unwrap()`' means nothing: the practical sense is that one *should* use it to pass iterators from `partition` and `span` into a `for` loop, at least on the first iterator.

I hope it does not sound as 'I like this wording and don't like that wording'. If my idea is not clear, let's discuss it voicely.